go/enclave/components/batch_executor.go

```diff
@@ -67,12 +67,12 @@ func NewBatchExecutor(
 	}
 }
 
-// payL1Fees - this function modifies the state db according to the transactions contained within the batch context
+// estimateL1Fees - this function modifies the state db according to the transactions contained within the batch context
 // in order to subtract gas fees from the balance. It returns a list of the transactions that have prepaid for their L1
 // publishing costs.
-func (executor *batchExecutor) payL1Fees(stateDB *state.StateDB, context *BatchExecutionContext) (common.L2Transactions, common.L2Transactions) {
-	transactions := make(common.L2Transactions, 0)
-	freeTransactions := make(common.L2Transactions, 0)
+func (executor *batchExecutor) estimateL1Fees(stateDB *state.StateDB, context *BatchExecutionContext) (common.L2PricedTransactions, common.L2PricedTransactions) {
```

Comment:
also looks to me like this method is excluding transactions that don't have enough money to pay for the l1 cost. 
I think the name and comment should reflect its responsibilities a bit better